CryptoPkg/Test/UnitTest/Library/BaseCryptLib/RsaPkcs7Tests.c

```diff
@@ -324,15 +337,28 @@ TestVerifyRsaCertPkcs1SignVerify (
   Status = X509VerifyCert (TestCert, sizeof (TestCert), TestCACert, sizeof (TestCACert));
   UT_ASSERT_TRUE (Status);
 
+  TestStatus = UNIT_TEST_PASSED;
+Exit:
   //
   // Release Resources.
   //
-  RsaFree (RsaPubKey);
-  RsaFree (RsaPrivKey);
-  FreePool (Signature);
-  FreePool (Subject);
+  if (RsaPrivKey != NULL) {
+    RsaFree (RsaPrivKey);
+  }
 
-  return UNIT_TEST_PASSED;
+  if (RsaPubKey != NULL) {
+    RsaFree (RsaPubKey);
+  }
+
+  if (Signature != NULL) {
+    FreePool (Signature);
+  }
+
+  if (Subject != NULL) {

```

Comment:
As a general practice, I suggest freeing these in reverse order of the allocations. It doesn't seem to be an issue here, but doing so can help more naturally release buffers that may be dependent on each other.

For example, (1) a structure buffer is allocated with (2) a pointer to another buffer that is allocated. Free (2), then free (1).

Reply:
Yeah that's totally fair

Reply:
Done in PR commit https://github.com/Flickdm/mu_basecore/pull/1/commits/59d446c9d3451f0c212cbb994684129a40e3e8ac